docs/backend-system/core-services/01-index.md

```diff
@@ -65,10 +88,366 @@ createBackendPlugin({
+      remote: { reloadIntervalSeconds: 60 }
+    }),
+  ],
+});
+```
+
+## Logging
+
+It is common for your plugins to be able to use the logger. This logger is bound to your plugin, so that you will get nice messages with the plugin ID referenced in the log lines.
+
+### Using the service
+
+The following example shows how to get config in your `example` backend plugin and create a `warn` that will be printed nicely to the console.
+
+```ts
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+
+createBackendPlugin({
+  id: 'example',
+  register(env) {
+    env.registerInit({
+      deps: {
+        log: coreServices.logger,
+      },
+      async init({ log }) {
+        log.warn('Heres a nice log line thats a warning!');
+      },
+    });
+  },
+});
+```
+
+## Cache
+
+There's a core service provided with the backend system that can be used to interact with a cache in your plugins. This cache is bound to your plugin too, so that you will only set and get values in your plugins namespace.
+
+### Using the service
+
+The following example shows how to get a cache client in your `example` backend plugin and `set` and `get` values from the cache.
+
+```ts
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+
+createBackendPlugin({
+  id: 'example',
+  register(env) {
+    env.registerInit({
+      deps: {
+        cache: coreServices.cache,
+      },
+      async init({ cache }) {
+        const { key, value } = { key: 'test:key', value: 'bob' };
+        await cache.set(key, value, { ttl: 1000 });
+
+        // .. some other stuff.
+
+        await cache.get(key); // 'bob'
+      },
+    });
+  },
+});
+```
+
+## Database
+
+Interacting with databases inside your plugin is something that is quite common, and we provide a `PluginDatabaseManager` part of the core services that you can get `knex` client hooked up to your database which is configured in `app-config.yaml`.
+
+If there's no config provided in `backend.database` then you will automatically get a simple in memory `sqlite3` client for your plugin.
+
+These `PluginDatabaseManager`s are scoped per plugin too, so that table names do not conflict across plugins either.
+
+### Using the service
+
+The following example shows how to get a `PluginDatabaseManager` in your `example` backend plugin and get a `client` for interacting with the database and running some migrations from a `migrationsDir` for your plugin.
+
+```ts
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+
+createBackendPlugin({
+  id: 'example',
+  register(env) {
+    env.registerInit({
+      deps: {
+        database: coreServices.database,
+      },
+      async init({ database }) {
+        const client = database.getClient();
+
+        if (!database.migrations?.skip) {
+          await client.migrate.latest({
+            directory: migrationsDir,
```

Comment:
It's really nice that you show this. But I'm wondering if it's valuable to actually show the whole resolvePackagePath dance at the same time then, since it's a bit tricky to make certain you are getting the right path both locally and in prod. Eg https://github.com/backstage/backstage/blob/5b3948a4690fd168364a53c715401fcf6bb62c96/plugins/search-backend-module-pg/src/database/DatabaseDocumentStore.ts#L30

Maybe in a follow-up PR then, since that entails also showing where to put the migrations dir and adding it to the `"files"` section in the package.json as well ...

Could be an entire "Migrations" subsection actually, but again, in a later PR